### ISSUE TITLE
Markdown list rendering problem when the list item contains equations

### ISSUE THREAD

**Issue by panqiincs** [theme-next/hexo-theme-next#889]
### I agree and want to create new issue 

- [x] Yes, I was on [Hexo Docs page](https://hexo.io/docs/), especially on [Templates](https://hexo.io/docs/templates.html), [Variables](https://hexo.io/docs/variables.html), [Helpers](https://hexo.io/docs/helpers.html) and [Troubleshooting](https://hexo.io/docs/troubleshooting.html).
- [x] Yes, I was on [NexT Documentation Site](http://theme-next.org/docs/).
- [x] And yes, I already searched for current [issues](https://github.com/theme-next/hexo-theme-next/issues?utf8=%E2%9C%93&q=is%3Aissue) and this is not help to me.

***

### Expected behavior 

Open Mathjax of a post by writing `mathjax: true`, all markdown list items display properly.

### Actual behavior 

Open Mathjax of a post by writing `mathjax: true`, when the item contains equations, it does not display properly

For exmaple, the following markdown:

```md
Please check the following items:

+ The equation $E=m^2$
+ No equations

and more:

1. I have a symbol $x$
2. I don't have a symbol
3. I also have a symbol $y$
```

The result is:

![Screenshot_20190528_113323](https://user-images.githubusercontent.com/3938330/58449054-79067d00-813c-11e9-8910-0c027aa6a8ef.png)

### Steps to reproduce the behavior 
1. Open Mathjax for a post by writing `mathjax: true` at the beginning
2. Write equation in list items
3. Generate page and check

* Link to demo site with this bug: N/A
* Link(s) to source code or any usefull link(s): N/A

### Node.js and NPM Information

```
v11.15.0
6.9.0
```

### Package dependencies Information

```
{
  "name": "hexo-site",
  "version": "0.0.0",
  "private": true,
  "hexo": {
    "version": "3.8.0"
  },
  "dependencies": {
    "hexo": "^3.8.0",
    "hexo-deployer-git": "^1.0.0",
    "hexo-generator-archive": "^0.1.5",
    "hexo-generator-category": "^0.1.3",
    "hexo-generator-feed": "^1.2.2",
    "hexo-generator-index": "^0.2.1",
    "hexo-generator-sitemap": "^1.2.0",
    "hexo-generator-tag": "^0.2.0",
    "hexo-related-popular-posts": "^3.0.5",
    "hexo-renderer-ejs": "^0.3.1",
    "hexo-renderer-marked": "^0.3.2",
    "hexo-renderer-stylus": "^0.3.3",
    "hexo-server": "^0.3.3"
  }
}
```

#### Hexo version

```
hexo: 3.8.0
hexo-cli: 1.1.0
os: Linux 4.19.45-1-MANJARO linux x64
node: 11.15.0
v8: 7.0.276.38-node.19
uv: 1.29.1
zlib: 1.2.11
brotli: 1.0.7
ares: 1.15.0
modules: 67
nghttp2: 1.36.0
napi: 4
llhttp: 1.1.1
http_parser: 2.8.0
openssl: 1.1.1b
cldr: 35.1
icu: 64.2
tz: 2019a
unicode: 12.1
```

#### Hexo Configuration

```yml

```

### NexT Information

**NexT Version:**

- [ ] Latest Master branch
- [x] Latest Release version
- [ ] Old version

**NexT Scheme:**

- [ ] All schemes
- [ ] Muse
- [ ] Mist
- [x] Pisces
- [ ] Gemini

#### NexT Configuration:

```yml
math:
  enable: true

  # Default (true) will load mathjax / katex script on demand.
  # That is it only render those page which has `mathjax: true` in Front-matter.
  # If you set it to false, it will load mathjax / katex srcipt EVERY PAGE.
  per_page: true

  engine: mathjax
  #engine: katex

  # hexo-renderer-pandoc (or hexo-renderer-kramed) needed to full MathJax support.
  mathjax:
    cdn: //cdn.jsdelivr.net/npm/mathjax@2/MathJax.js?config=TeX-AMS-MML_HTMLorMML
    #cdn: //cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-MML-AM_CHTML

    # See: https://mhchem.github.io/MathJax-mhchem/
    #mhchem: //cdn.jsdelivr.net/npm/mathjax-mhchem@3
    #mhchem: //cdnjs.cloudflare.com/ajax/libs/mathjax-mhchem/3.3.0

  # hexo-renderer-markdown-it-plus (or hexo-renderer-markdown-it with markdown-it-katex plugin) needed to full Katex support.
  katex:
    cdn: //cdn.jsdelivr.net/npm/katex@0/dist/katex.min.css
    #cdn: //cdnjs.cloudflare.com/ajax/libs/KaTeX/0.7.1/katex.min.css

    copy_tex:
      # See: https://github.com/KaTeX/KaTeX/tree/master/contrib/copy-tex
      enable: false
      copy_tex_js: //cdn.jsdelivr.net/npm/katex@0/dist/contrib/copy-tex.min.js
      copy_tex_css: //cdn.jsdelivr.net/npm/katex@0/dist/contrib/copy-tex.min.css
```

**Comment by 1v9** [comment]
Thanks to Bugjax 😂

https://github.com/theme-next/hexo-theme-next/commits/master/source/css/_common/components/third-party/math.styl
https://github.com/theme-next/hexo-theme-next/commits/master/layout/_third-party/math/mathjax.swig

![1.gif](https://i.loli.net/2019/05/29/5ced7989c27c145004.gif)

**Comment by panqiincs** [comment]
@1v9 You mean it is a mathjax bug and cannot fix it in NeXT?

**Comment by JiangTJ** [comment]
Sure this is a problem that needs to be fixed. But it seems a little difficult.

**Comment by 1v9** [comment]
> @1v9 You mean it is a mathjax bug and cannot fix it in NeXT?

No but it's complicated. See #645 #658 #669 #721.

**Comment by stevenjoezhang** [comment]
https://stackoverflow.com/questions/21769248/list-with-nested-overflow-x-hidden-hides-list-counter-point-why-is-this-a-b

**Comment by stevenjoezhang** [comment]
Duplicate of https://github.com/theme-next/hexo-theme-next/pull/745
> This problem is really annoying because it is a browser bug instead of design error.

We can't fix it.

**Comment by JiangTJ** [comment]
@stevenjoezhang How about remove `overflow` in `li`? Strange issue...

**Comment by stevenjoezhang** [comment]
Removing `overflow` in `li` is not a good idea: https://github.com/theme-next/hexo-theme-next/issues/645

**Comment by JiangTJ** [comment]
![image](https://user-images.githubusercontent.com/15902347/58545800-991a6700-8236-11e9-94d5-d7a63698f64e.png)
Is it possible to add to `ul`?

**Comment by JiangTJ** [comment]
It seems that `overflow` and `li` are conflicting in browser now, and should not be used at the same time.

**Comment by JiangTJ** [comment]
![image](https://user-images.githubusercontent.com/15902347/58546280-93715100-8237-11e9-9def-a826656ce639.png)
![image](https://user-images.githubusercontent.com/15902347/58546287-979d6e80-8237-11e9-8578-35a4134f9ee6.png)
Or wrap with `div` ...

**Comment by stevenjoezhang** [comment]
Fixed in #1004
